src/components/details/types.d.ts

```diff
@@ -0,0 +1,23 @@
+type PokemonStats = {
+    name: string,
+    url: string,
+    height: number,
+    weight: number,
+    abilities: Ability[],
+    moves: Move[],
+}
+
+interface Ability {
```

Comment:
How come it is Ability { ability { ?

Reply:
Because in the JSON response, the ability object has another nested ability object, with other members on the top level, e.g. 

```
"abilities": [
        {
            "ability": {
                "name": "overgrow",
                "url": "https://pokeapi.co/api/v2/ability/65/"
            },
            "is_hidden": false,
            "slot": 1
        },
        {
            "ability": {
                "name": "chlorophyll",
                "url": "https://pokeapi.co/api/v2/ability/34/"
            },
            "is_hidden": true,
            "slot": 3
        }
    ],
```

instead of 

```
"abilities": [
        {
                "name": "overgrow",
                "url": "https://pokeapi.co/api/v2/ability/65/"
        },
        {
                "name": "chlorophyll",
                "url": "https://pokeapi.co/api/v2/ability/34/"
        }
    ],
```